<html>
<head><meta charset="utf-8"><title>running script in batch mode · python-questions · Zulip Chat Archive</title></head>
<h2>Stream: <a href="http://127.0.0.1:4000/html/stream/10-python-questions/index.html">python-questions</a></h2>
<h3>Topic: <a href="http://127.0.0.1:4000/html/stream/10-python-questions/topic/running.20script.20in.20batch.20mode.html">running script in batch mode</a></h3>

<hr>


<head><link href="http://127.0.0.1:4000/style.css" rel="stylesheet"></head>

<a name="55101"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/10-python-questions/topic/running%20script%20in%20batch%20mode/near/55101" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jean-Francois Lamarque <a href="http://127.0.0.1:4000/html/stream/10-python-questions/topic/running.20script.20in.20batch.20mode.html#55101">(May 18 2022 at 00:16)</a>:</h4>
<p>I have a bash script that I submit to qsub on casper.  Before the update last week, it was working great but now it is not working at all. I am getting the error message</p>
<p>Warning: no access to tty (Bad file descriptor).<br>
Thus no job control in this shell.<br>
Now using NPL virtual environment at path:<br>
    /glade/u/apps/dav/opt/python/3.7.9/gnu/9.1.0/pkg-library/20201220</p>
<p>Use deactivate to remove NPL from environment</p>
<p>The script is /glade/u/home/lamar/Python/CMIP6_analysis/PM2.5/run_health.bash</p>
<p>Thanks!</p>



<a name="55107"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/10-python-questions/topic/running%20script%20in%20batch%20mode/near/55107" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Matt Long <a href="http://127.0.0.1:4000/html/stream/10-python-questions/topic/running.20script.20in.20batch.20mode.html#55107">(May 18 2022 at 15:22)</a>:</h4>
<p>Are you sure it's not working?  I think this part of the warning:</p>
<div class="codehilite" data-code-language="Bash"><pre><span></span><code>Warning: no access to tty <span class="o">(</span>Bad file descriptor<span class="o">)</span>.
Thus no job control <span class="k">in</span> this shell.
</code></pre></div>
<p>may just be telling you that there's no input device. </p>
<p>And this part</p>
<div class="codehilite" data-code-language="Bash"><pre><span></span><code>Now using NPL virtual environment at path:
/glade/u/apps/dav/opt/python/3.7.9/gnu/9.1.0/pkg-library/20201220

Use deactivate to remove NPL from environment
</code></pre></div>
<p>is what <code>ncar_pylib</code> prints out.</p>



<a name="55181"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/10-python-questions/topic/running%20script%20in%20batch%20mode/near/55181" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jean-Francois Lamarque <a href="http://127.0.0.1:4000/html/stream/10-python-questions/topic/running.20script.20in.20batch.20mode.html#55181">(May 19 2022 at 16:28)</a>:</h4>
<p>Got the answer from CISL.  ncar_pylib is deprecated and needs to be replaced by the following</p>
<p>module load ncarenv<br>
module load conda<br>
conda activate npl</p>
<h1>python compute_health_5years_omp_batch.py</h1>



<hr><p>Last updated: May 16 2025 at 17:14 UTC</p>
</html>